qiskit/algorithms/observables_evaluator.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Evaluator of auxiliary operators for algorithms."""
+from __future__ import annotations
+
+from typing import Tuple, List
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.opflow import (
+    PauliSumOp,
+)
+from . import AlgorithmError
+from .list_or_dict import ListOrDict
+from ..primitives import EstimatorResult, BaseEstimator
+from ..quantum_info.operators.base_operator import BaseOperator
+
+
+def eval_observables(
+    estimator: BaseEstimator,
+    quantum_state: QuantumCircuit,
+    observables: ListOrDict[BaseOperator | PauliSumOp],
+    threshold: float = 1e-12,
+) -> ListOrDict[Tuple[complex, complex]]:
+    """
+    Accepts a sequence of operators and calculates their expectation values - means
+    and standard deviations. They are calculated with respect to a quantum state provided. A user
+    can optionally provide a threshold value which filters mean values falling below the threshold.
+
+    Args:
+        estimator: An estimator primitive used for calculations.
+        quantum_state: An unparametrized quantum circuit representing a quantum state that
+            expectation values are computed against.
+        observables: A list or a dictionary of operators whose expectation values are to be
+            calculated.
+        threshold: A threshold value that defines which mean values should be neglected (helpful for
+            ignoring numerical instabilities close to 0).
+
+    Returns:
+        A list or a dictionary of tuples (mean, standard deviation).
+
+    Raises:
+        ValueError: If a ``quantum_state`` with free parameters is provided.
+        AlgorithmError: If a primitive job is not successful.
+    """
+
+    if (
+        isinstance(quantum_state, QuantumCircuit)  # Statevector cannot be parametrized
```

Comment:
State is a bit more general, where "vector" might be confusing since this is just a circuit 🙂
```suggestion
        isinstance(quantum_state, QuantumCircuit)  # State cannot be parametrized
```